db/migrations.go

```diff
@@ -9,34 +9,71 @@ import (
 )
 
 type localMigration struct {
-	Name  string
-	Query string
+	Name  	  string
+	Condition []string
+	Query	  []string
 }
 
 var migrations = []localMigration{
 	{
-		Name:  "2020_11_03_04_42_SetDefaultDownloadStatus",
-		Query: "update podcast_items set download_status=2 where download_path!='' and download_status=0",
+		Name:  	   "2020_11_03_04_42_SetDefaultDownloadStatus",
+		Condition: []string{},
+		Query:     []string{
+			"update podcast_items set download_status=2 where download_path!='' and download_status=0",
+		},
+	},
+	{
+		Name:  "2021_06_01_00_00_ConvertFileNameFormat",
+		Condition: []string{
+			"SELECT COUNT(*) > 0 FROM (SELECT name FROM pragma_table_info('settings') where name is 'append_date_to_file_name');",
+			"SELECT COUNT(*) > 0 FROM (SELECT name FROM pragma_table_info('settings') where name is 'append_episode_number_to_file_name');",
+		},
+		Query: []string{
+			"UPDATE settings SET file_name_format = CASE WHEN append_date_to_file_name AND append_episode_number_to_file_name THEN '%EpisodeNumber%-%EpisodeDate%-%EpisodeTitle%' WHEN append_date_to_file_name THEN '%EpisodeDate%-%EpisodeTitle%' WHEN append_episode_number_to_file_name THEN '%EpisodeNumber%-%EpisodeTitle%' ELSE '%EpisodeTitle%' END",
+// sqlite3 v3.35.0 supports DROP COLUMN
+// 			"ALTER TABLE settings DROP COLUMN append_episode_number_to_file_name",
+// 			"ALTER TABLE settings DROP COLUMN append_date_to_file_name",
+		},
```

Comment:
ConvertFileNameFormat needs conditions to ensure it doesn't try to convert the old "append" fields to the new "file name format" if those fields don't exist in the database. Ideally multiple Query statements would be used to drop the old "append" columns, but that's not supported until sqlite3 v3.35. The alternative is recreating the table, but that seems excessive.
Are there other ways to accomplish this?